packages/dds/map/src/directory.ts

```diff
@@ -1734,6 +1857,38 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
                 if (pendingMessageIds.length === 0) {
                     this.pendingSubDirectories.delete(op.subdirName);
                 }
+                if (op.type === "deleteSubDirectory") {
+                    const count = this.pendingDeleteSubDirectoriesCount.get(op.subdirName);
+                    assert(count !== undefined && count > 0, "should have record for delete op");
+                    this.pendingDeleteSubDirectoriesCount.set(op.subdirName, count - 1);
+                    if (count === 1) {
+                        this.pendingDeleteSubDirectoriesCount.delete(op.subdirName);
+                    }
+                }
+            } else if (op.type === "deleteSubDirectory") {
+                // If this is remote delete op and we have keys in this subDirectory, then we need to delete these
+                // keys except the pending ones as they will be sequenced after this delete.
+                const subDirectory = this._subdirectories.get(op.subdirName);
+                if (subDirectory) {
+                    subDirectory.clearExceptPendingKeys();
+                    // In case of remote delete op, we need to reset the creation seq number and client ids of
+                    // creators as the previous directory is getting deleted and we will initialize again when
+                    // we will receive op for the create again.
```

Comment:
This comment does not match my mental model. I'd think that any operations for old instance of this sub-directory should be dropped on the floor.